doc/source/programs/gdal_raster_pixelinfo.rst

```diff
@@ -0,0 +1,238 @@
+.. _gdal_raster_pixelinfo:
+
+================================================================================
+``gdal raster pixelinfo``
+================================================================================
+
+.. versionadded:: 3.11
+
+.. only:: html
+
+    Return information on a pixel of a raster dataset
+
+.. Index:: gdal raster pixelinfo
+
+Synopsis
+--------
+
+.. program-output:: gdal raster pixelinfo --help-doc
+
+Description
+-----------
+
+:program:`gdal raster pixelinfo` provide a mechanism to query information about
+a pixel given its location in one of a variety of coordinate systems.
+
+It supports outputting either as GeoJSON or CSV.
```

Comment:
Why not create OGR features and write to any format? I can see wanting this output to Parquet or Postgres

Reply:
The tricky part is the interactive mode where you do "gdal raster pixelinfo my.tif"  and can input input coordinates interactively. We want a text output in that mode. I would say that binary mode only makes sense if we accept input coordinates from a vector dataset, so as a later enhancement